SensusService/DataStores/DataStore.cs

```diff
@@ -378,26 +404,23 @@ public DataStore Copy()
                 PreserveReferencesHandling = PreserveReferencesHandling.None,
                 ReferenceLoopHandling = ReferenceLoopHandling.Ignore,
                 TypeNameHandling = TypeNameHandling.All,
-                ConstructorHandling = ConstructorHandling.AllowNonPublicDefaultConstructor
+                ConstructorHandling = ConstructorHandling.AllowNonPublicDefaultConstructor,
             };
 
-            DataStore copy = null;
             try
             {
                 SensusServiceHelper.Get().FlashNotificationsEnabled = false;
-                copy = JsonConvert.DeserializeObject<DataStore>(JsonConvert.SerializeObject(this, settings), settings);
+                return JsonConvert.DeserializeObject<DataStore>(JsonConvert.SerializeObject(this, settings), settings);
             }
             catch (Exception ex)
             {
-                SensusServiceHelper.Get().Logger.Log("Failed to copy data store:  " + ex.Message, LoggingLevel.Normal, GetType());
-                copy = null;
+                SensusServiceHelper.Get().Logger.Log($"Failed to copy data store:  {ex.Message}", LoggingLevel.Normal, GetType());
+                throw;
```

Comment:
No. The old code still failed it just hid the serialize errors. By returning null we were causing a null reference exception to occur further down the line. Choosing to not throw wasn't preventing errors it was simply hiding helpful feedback when the error occurred, so I told Wes to throw.